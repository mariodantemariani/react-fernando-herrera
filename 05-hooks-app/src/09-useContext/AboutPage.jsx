export const AboutPage = () => {
  return (
    <>
      <h1>About APp</h1>
      <hr />
    </>
  );
};
