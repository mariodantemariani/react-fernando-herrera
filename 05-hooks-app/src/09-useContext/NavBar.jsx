import { Link, NavLink } from "react-router-dom";

export const NavBar = () => {
  return (
    <>
      <nav className="navbar navbar-expand-lg navbar-dark bg-dark rounded-3">
        <div className="container-fluid">
          <Link className="navbar-brand" href="#">
            usContext
          </Link>
          <div className="collapse navbar-collapse" id="navbarNav">
            <ul className="navbar-nav">
              <NavLink
                className={(args) =>
                  `nav-link ${args.isActive ? "active" : ""}`
                }
                to="/"
              >
                Home
              </NavLink>
              {/* <li className="nav-item">
                <Link className="nav-link" href="#">
                  Home
                </Link>
              </li> */}
              <NavLink
                className={(args) =>
                  `nav-link ${args.isActive ? "active" : ""}`
                }
                to="/about"
              >
                About
              </NavLink>

              <NavLink
                className={(args) =>
                  `nav-link ${args.isActive ? "active" : ""}`
                }
                to="/login"
              >
                Login
              </NavLink>

              {/* <li className="nav-item ">
                <Link className="nav-link active" href="#">
                  Features
                </Link>
              </li> */}
            </ul>
          </div>
        </div>
      </nav>
    </>
  );
};
