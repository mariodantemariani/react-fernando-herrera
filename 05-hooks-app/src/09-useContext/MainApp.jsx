import { Navigate, Route, Routes } from "react-router-dom";

import { AboutPage, LoginPage, HomePage } from ".";

import { NavBar } from "../09-useContext/NavBar";
import { UserProvider } from "../09-useContext/context/UserProvider";

export const MainApp = () => {
  return (
    <UserProvider>
      {/* <h1>Main APP</h1> */}
      <NavBar />
      <hr />
      <Routes>
        <Route path="/" element={<HomePage />} />
        <Route path="about" element={<AboutPage />} />
        <Route path="login" element={<LoginPage />} />
        {/* <Route path="*" element={<HomePage />} /> */}
        <Route path="/*" element={<Navigate to="/about" />} />
      </Routes>
    </UserProvider>
  );
};
