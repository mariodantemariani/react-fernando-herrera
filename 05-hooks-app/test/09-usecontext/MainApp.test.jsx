import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import { MainApp } from "../../src/09-useContext/MainApp";

describe("Pruebas en <MainApp />", () => {
  test("should be the same that snapShot", () => {
    const { container } = render(
      <MemoryRouter>
        <MainApp />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  test("debe de mostrar el HomePage", () => {
    render(
      <MemoryRouter>
        <MainApp />
      </MemoryRouter>
    );

    expect(screen.getByText("Home")).toBeTruthy();
    //screen.debug();
  });

  test("debe de mostrar el LoginPage", () => {
    render(
      <MemoryRouter initialEntries={["/login"]}>
        <MainApp />
      </MemoryRouter>
    );

    expect(screen.getByText("Login")).toBeTruthy();
    //screen.debug();
  });
});
