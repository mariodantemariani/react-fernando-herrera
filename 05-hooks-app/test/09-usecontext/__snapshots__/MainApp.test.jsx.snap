// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MainApp /> should be the same that snapShot 1`] = `
<div>
  <nav
    class="navbar navbar-expand-lg navbar-dark bg-dark rounded-3"
  >
    <div
      class="container-fluid"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        usContext
      </a>
      <div
        class="collapse navbar-collapse"
        id="navbarNav"
      >
        <ul
          class="navbar-nav"
        >
          <a
            aria-current="page"
            class="nav-link active"
            href="/"
          >
            Home
          </a>
          <a
            class="nav-link "
            href="/about"
          >
            About
          </a>
          <a
            class="nav-link "
            href="/login"
          >
            Login
          </a>
        </ul>
      </div>
    </div>
  </nav>
  <hr />
  <h1>
    Home APP 
  </h1>
  <hr />
  <pre
    aria-label="pre"
  />
</div>
`;
